### PR TITLE
test(accordion): migrate component tests to playwright

### PR DESCRIPTION
**Pull request by robinzigmond**
### Proposed behaviour

The Accordion component should have in-browser tests using the Playwright framework

### Current behaviour

Accordion currently has Cypress tests instead of Playwright tests

### Checklist

- [X] Commits follow our style guide
- [X] Related issues linked in commit messages if required
- [X] Screenshots are included in the PR if useful
- [X] All themes are supported if required
- [X] Unit tests added or updated if required
- [X] Cypress automation tests added or updated if required
- [X] Playwright automation tests added or updated if required
- [X] Storybook added or updated if required
- [X] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [X] Typescript `d.ts` file added or updated if required
- [X] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

File: src/components/accordion/accordion.pw.tsx [hunk 1]
```diff
@@ -0,0 +1,735 @@
...
+        "aria-expanded",
+        String(state)
+      );
+      await expect(accordionTitleContainer(page)).toBeVisible();
+    });
+  });
+
+  ([
+    ["white", "rgb(255, 255, 255)", "be.visible"],
+    ["transparent", "rgba(0, 0, 0, 0)", "be.visible"],
```
Comment (stephenogorman) [inline src/components/accordion/accordion.pw.tsx:330]: Should third parameter "be.visible" be here?

Comment (robinzigmond) [reply src/components/accordion/accordion.pw.tsx:330]: good spot, thanks. Looks like I forgot to remove this from the Cypress test even though we're not using it.

File: src/components/accordion/accordion.pw.tsx [hunk 2]
```diff
@@ -0,0 +1,735 @@
...
+      "rgba(0, 0, 0, 0) solid 3px"
+    );
+  });
+
+  test("should have the expected styling when the focusRedesignOptOut is true", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<AccordionComponent />, {
+      hooksConfig: { focusRedesignOptOut: true },
```
Comment (ZhuoyuJin) [inline src/components/accordion/accordion.pw.tsx:68]: Should we import HooksConfig from playwright and use it here like: "await mount<HooksConfig>(<AccordionComponent />"

File: src/components/accordion/accordion.pw.tsx [hunk 3]
```diff
@@ -0,0 +1,735 @@
...
+    await expect(accordionTitleContainer(page)).toContainText("Open");
+  });
+});
+
+test.describe("should render Accordion Grouped component", () => {
+  test("should move through all grouped accordions using ArrowDown key and check focus when focusRedesignOptOut is true", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<AccordionGroupComponent />, {
```
Comment (ZhuoyuJin) [inline src/components/accordion/accordion.pw.tsx:437]: same here

File: src/components/accordion/accordion.pw.tsx [hunk 4]
```diff
@@ -0,0 +1,735 @@
...
+    );
+    await expect(accordionTitleContainer(page).nth(2)).toBeVisible();
+  });
+
+  test("should move to the last grouped accordion using End key and check it is focused when focusRedesignOptOut is true", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<AccordionGroupComponent />, {
+      hooksConfig: { focusRedesignOptOut: true },
```
Comment (ZhuoyuJin) [inline src/components/accordion/accordion.pw.tsx:487]: same here

File: src/components/accordion/accordion.pw.tsx [hunk 5]
```diff
@@ -0,0 +1,735 @@
...
+    );
+    await expect(accordionTitleContainer(page).nth(2)).toBeVisible();
+  });
+
+  test("should move to the first grouped accordion using Home key and check it is focused when focusRedesignOptOut is true", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<AccordionGroupComponent />, {
+      hooksConfig: { focusRedesignOptOut: true },
```
Comment (ZhuoyuJin) [inline src/components/accordion/accordion.pw.tsx:520]: same here

**Review by nineteen88 — APPROVED** [review]

**Review by stephenogorman — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Review by nineteen88 — APPROVED** [review]

**Pull request merged**